### ISSUE TITLE
Map [control key + left click] to [right click] on OSX.

### ISSUE THREAD

**Issue by pk-nb** [aseprite/aseprite#938]
There are [multiple ways to right click on OS X](https://support.apple.com/kb/PH18768?locale=en_US) with both a trackpad and mouse. One method, a click while holding down the Control key, works across the system but unfortunately not in Aseprite. Other right-click input methods such as a double finger click on the trackpad do work. However some features like shading are difficult to use on a trackpad without the Control key support.

<img width="641" alt="screen shot 2016-01-17 at 10 07 18 pm" src="https://cloud.githubusercontent.com/assets/1911028/12383825/b70eb83e-bd67-11e5-8a1a-59995492f61f.png">

Of course, this may be low priority [with the effort to abstract to another view backends](https://github.com/aseprite/aseprite/issues/139) but curious if this is possible to fix quickly with the current code base. I'm still pretty unfamiliar with the repo—[would `system.m` be the appropriate place](https://github.com/aseprite/aseprite/blob/master/src/allegro/src/macosx/system.m#L278) to map to a right click with the control modifier? I'm assuming other events such as click drag would need the modification as well.  Let me know if I can help—thank you for all the work on this great app!

**Comment by dacap** [comment]
Hi @pk-nb, the Allegro 4 backend (e.g. the `system.m` file you linked) is deprecated. v1.1 series will be the latest ones with this backend. From v1.2 we will use the Skia backend (which is already working in some ways). You can find a better trackpad support on that branch. And some code to receive `NSEvent`s and convert them to `she::Event`s is inside [OSXView](https://github.com/aseprite/aseprite/blob/master/src/she/osx/view.mm#L76). That is the code that we'll maintain/improve from now on.

Anyway there are no instructions on yet to compile the Skia port correctly, but you should use the `USE_SKIA_BACKEND=ON` and `USE_ALLEG4_BACKEND=OFF`, and point the `SKIA_DIR` variable to a correctly compiled version of [Skia library](http://skia.org/) (branch `m48` if I remember correctly).

**Comment by pk-nb** [comment]
Thanks for the quick response @dacap. I believe I have Skia setup correctly with your comment here (`cmake -G "Unix Makefiles"` correctly finds the built Skia lib) but am still having build issues on OS X 10.11 El Capitan. The OS X `she` files are failing, presumably due to legacy OS X version differences.

```
$ make she
[  0%] Built target modp_b64
[ 37%] Built target base-lib
[ 37%] Built target fixmath-lib
[ 75%] Built target pixman
[ 87%] Built target gfx-lib
[ 87%] Building CXX object src/she/CMakeFiles/she.dir/osx/app.mm.o
error: -fobjc-arc is not supported on platforms using the
      legacy runtime
make[3]: *** [src/she/CMakeFiles/she.dir/osx/app.mm.o] Error 1
make[2]: *** [src/she/CMakeFiles/she.dir/all] Error 2
make[1]: *** [src/she/CMakeFiles/she.dir/rule] Error 2
make: *** [she] Error 2
```

I thought installing old 10.4 SDK was only necessary for the Allegro 4 backend, but tried it anyway after running into this issue. Unfortunately the SDK will not install (incompatible with this version of OS X). Sorry to bug you with build issues, but I'm happy to help / contribute if I can get this working.

**Comment by dacap** [comment]
This issue was fixed in #438 with f5c43267fca6770611b842d97410e5baadcda8be for some widgets. But the sprite editor doesn't interpret the Ctrl+click as right-click (because Ctrl is already a modifier to change to the Hand tool).
